src/software/backend/simulation/physics_ball_test.cpp

```diff
@@ -15,8 +15,8 @@ TEST(PhysicsBallTest, test_get_ball_with_timestamp)
     auto physics_ball = PhysicsBall(world, ball_parameter);
     auto ball         = physics_ball.getBallWithTimestamp(Timestamp::fromSeconds(1.1));
 
-    EXPECT_TRUE(ball_parameter.position().isClose(ball.position(), 1e-7));
-    EXPECT_TRUE(ball_parameter.velocity().isClose(ball.velocity(), 1e-7));
+    EXPECT_TRUE(ball_parameter.position().isClose(ball.position(), 1e-4));
+    EXPECT_TRUE(Point(ball_parameter.velocity()).isClose(Point(ball.velocity()), 1e-7));
```

Comment:
Should this be tested with `EXPECT_EQ` instead with the `Vector`'s `==` operator? Seems unnecessarily convoluted to have to convert the two `Vector`s to `Point`s just to compare them. If we want to be able to control the value of `EPSILON` for comparing `Vector`'s, maybe `Vector` class should have its own method for it. Also applies to other places like `physics_field_test.cpp`.

Reply:
Yeah I ran into several of these cases, maybe something like ```Vector::almostSame(const Vector& other)``` since I'm not sure if it makes sense for two vectors to be "close" to one another.

Reply:
Yes I believe the simulation stores things internally as floats and accuracy is lost during the simulation steps (because things can't always be solved perfectly), which is why I had do do custom comparisons like this.

`almostEqual` or a vector equivalent to `isClose` makes sense.

Reply:
I dislike both `almostEqual` and `isClose`, because they are both pretty subjective. I would be fine with implementing the `==` operator to check if two vectors are equal to within machine precision though? If we want to check if something is close to within `float` tolerance, we should check if both the `x` and `y` components are equal using `std::numeric_limits<float>::epsilon`

Reply:
@garethellis0 for these tests then do you think we should subtract the vectors and take the difference (rather than use a function that determines "closeness")?

Eg. `EXPECT_LT((ball_parameter.velocity() - ball.velocity()).len(), 1e-6);`

Reply:
Yeah, I'm totally fine with that, main issue I had was having arbitrary methods on the `Point`/`Vector` classes themselves to check "almost equal" or "is close"

Reply:
Changed tests that convert ```Vector```s to ```Point```s THEN uses ```isClose(...)``` to comparing the vector difference to a small length instead (refer to Mathew's example above). This is in the latest update.